src/ui/components/shared/WorkspaceSettingsModal/WorkspaceMember.tsx

```diff
@@ -70,52 +80,73 @@ function WorkspaceMemberRoles({
   member: WorkspaceUser;
   onClick: () => void;
 }) {
-  const role: WorkspaceUserRole = getMemberRole(member);
+  const roles: WorkspaceUserRole[] = getMemberRoles(member);
   const memberIsAdmin = getIsAdmin(member);
   const { updateWorkspaceMemberRole } = hooks.useUpdateWorkspaceMemberRole();
-  const [selectedRole, setSelectedRole] = useState<WorkspaceUserRole>(role);
-
-  useEffect(() => setSelectedRole(role), [role]);
+  const [loading, setLoading] = useState(false);
 
-  const selectRole = (updated: WorkspaceUserRole) => {
-    onClick();
-    let roles: WorkspaceUserRole[];
-
-    if (updated !== "admin") {
-      roles = [updated];
-      if (memberIsAdmin) {
-        roles.push("admin");
-      }
-    } else {
-      roles = [role];
-      if (!memberIsAdmin) {
-        roles.push("admin");
-      }
-    }
-
-    setSelectedRole(updated);
+  const updateRoles = (roles: string[]) => {
+    setLoading(true);
 
     updateWorkspaceMemberRole({
       variables: {
         id: member.membershipId,
         roles,
       },
-    }).catch(e => {
-      console.error(e);
-      setSelectedRole(selectedRole);
-    });
+    })
+      .catch(e => {
+        console.error(e);
+      })
+      .finally(() => {
+        setLoading(false);
+        onClick();
+      });
+  };
+
+  const selectViewerRole = (selected: boolean) => {
+    // when switching to viewer role, we remove debugger
+    if (selected) {
+      updateRoles(roles.filter(r => r !== "debugger"));
```

Comment:
Correct because the user should always have the viewer role. I can add a safety valve to ensure it's here though.

Reply:
I see. It seems like we're representing our (actual) model in a weird way with this UI. I wonder if it would be better if we did something like...
![Screenshot 2024-03-05 at 3 42 16 PM](https://github.com/replayio/devtools/assets/29597/44a73663-027b-4335-8d29-5e49b4e39d7b)

That being said, I realize that's a more significant change and I won't block this PR on it :)

Reply:
Pushed a commit that ensures viewer is there when that role is selected

Reply:
I didn't have your last two comments yet when I last responded so I'll add: Yes! We should improve this. I don't know if there's a DES ticket for this yet but I'll make one and Jon can cancel it if it's a dupe.